lte/gateway/docker/mme/Dockerfile.ubuntu18.04

```diff
@@ -116,94 +123,131 @@ RUN git clone --recurse-submodules -b v1.15.0 https://github.com/grpc/grpc && \
         -DgRPC_CARES_PROVIDER=package \
         -DgRPC_SSL_PROVIDER=package \
         -DCMAKE_BUILD_TYPE=Release \
-        ../.. && \
+        .. && \
     make -j`nproc` && \
-    make install
+    make install && \
+    cd .. && rm -Rf _build
+
 
 ##### Prometheus CPP
-RUN git clone https://github.com/jupp0r/prometheus-cpp.git && \
-    cd prometheus-cpp && \
+RUN cd prometheus-cpp && \
+    # Moved git clone https://github.com/jupp0r/prometheus-cpp.git && \
     git checkout d8326b2bba945a435f299e7526c403d7a1f68c1f && \
     git submodule init && git submodule update && \
     mkdir _build && \
     cd _build/ && \
     cmake .. && \
     make -j`nproc` && \
-    make install
+    make install && \
+    cd .. && rm -Rf _build
 
 ##### Redis CPP
-RUN git clone https://github.com/cpp-redis/cpp_redis.git && \
-    cd cpp_redis && \
+RUN cd cpp_redis && \
+    # Moved git clone https://github.com/cpp-redis/cpp_redis.git && \
     git checkout bbe38a7f83de943ffcc90271092d689ae02b3489 && \
     git submodule init && git submodule update && \
-    mkdir build && cd build && \
+    mkdir _build && cd _build && \
     cmake .. -DCMAKE_BUILD_TYPE=Release && \
     make -j`nproc` && \
-    make install
+    make install && \
+    cd .. && rm -Rf _build
+
+##### NETTLE / gnutls
+RUN tar -xf nettle-2.5.tar.gz && \
+    # Moved wget https://ftp.gnu.org/gnu/nettle/nettle-2.5.tar.gz && \
+    cd nettle-2.5 && \
+    mkdir _build && \
+    cd _build/ && \
+    ../configure --disable-openssl --enable-shared --libdir=/usr/lib && \
+    make -j`nproc` && \
+    make install && \
+    ldconfig -v && \
+    cd / && \
+    rm -Rf nettle-2.5.tar.gz nettle-2.5/_build && \
+    # Moved wget https://www.gnupg.org/ftp/gcrypt/gnutls/v3.1/gnutls-3.1.23.tar.xz && \
+    tar xf gnutls-3.1.23.tar.xz && \
+    cd gnutls-3.1.23 && \
+    ./configure --with-libnettle-prefix=/usr --prefix=/usr && \
+    make -j`nproc` && \
+    make install && \
+    make clean 2>&1 > /dev/null && \
+    cd / && \
+    rm -Rf gnutls-3.1.23.tar.xz && \
+    ldconfig -v
 
 ##### liblfds
 # https://www.liblfds.org/mediawiki/index.php?title=r7.1.0:Building_Guide_(liblfds)
-RUN git clone https://github.com/liblfds/liblfds.git && \
-    cd liblfds/liblfds/liblfds7.1.0/liblfds710/build/gcc_gnumake/ && \
+RUN cd /liblfds/liblfds/liblfds7.1.0/liblfds710/build/gcc_gnumake/ && \
     make -j`nproc` && \
-    make ar_install
+    make ar_install && \
+    make clean
 
 ##### libgtpnl
 # review https://github.com/OPENAIRINTERFACE/openair-cn/blob/master/build/tools/build_helper.gtpnl
-RUN git clone https://git.osmocom.org/libgtpnl && \
-    cd libgtpnl && \
+RUN cd libgtpnl && \
+    # Moved git clone https://git.osmocom.org/libgtpnl && \
     git reset --hard 345d687 && \
     autoreconf -fi && \
     ./configure && \
     make -j`nproc` && \
     make install && \
+    make clean 2>&1 > /dev/null && \
```

Comment:
Would it make sense to use the `git clean` here as well?

Reply:
@LKreutzer 

I try to use `Makefile cleaning` tools wherever it is available and properly implemented.
If not, then I reverted to `git clean`.